client/a8c-for-agencies/components/items-dashboard/items-dataviews/index.tsx

```diff
@@ -23,26 +24,27 @@ const getIdByPath = ( item: object, path: string ) => {
 /**
  * Create an item column for the DataViews component
  * @param id
- * @param header
+ * @param label
  * @param displayField
  * @param getValue
  * @param isSortable
  * @param canHide
  */
 export const createItemColumn = (
 	id: string,
-	header: ReactNode,
+	label: ReactNode,
 	displayField: () => ReactNode,
 	getValue: () => undefined,
 	isSortable: boolean = false,
 	canHide: boolean = false
-): DataViewsColumn => {
+): Field< any > => {
 	return {
 		id,
 		enableSorting: isSortable,
 		enableHiding: canHide,
 		getValue,
-		header,
+		// @ts-expect-error -- Need to fix the label type upstream in @wordpress/dataviews to support React elements.
```

Comment:
@oandregal @ntsekouras I noticed that in a lot of places in .com, they use React elements as headers/labels for fields. It seems to work even though the `DataViews` type say it should be a string.

We should either update the type upstream in the Gutenberg repository. Or if it's mandatory to be a string, provide folks with an alternative way to render elements in the headers. (maybe by having both header and label)

Reply:
I agree it's sensible to allow consumers provide a component. Two valid use cases for this I've seen in this codebase are:

- render icons
- guided tour (attach a reference)

In my view, if the consumer provides a component it should still act as the "label" and the core affordances (sort, reorder fields, hide, add filter) should still work and be visible to interact with. For example, this codebase needs to implement a custom sort function to compensate the missing core affordances.

Reply:
Actually, the core controls in these DataViews are hidden because we're using `enableSorting` and `enableHiding`. So providing a custom sort is intentional.

My understanding is that these DataViews implementations are to be shared among dotcom, jetpack, etc. If the core tools are inconvenient, there should be design discussions to improve them but not provide a custom solution for each. I won't do that here in the interest of shipping the update, it can be a follow-up PR.

In relation to this, I've prepared a PR to a new `enableMoving` setting https://github.com/WordPress/gutenberg/pull/64646 to help clients fine-tune what a field can do.